src/libpriv/rpmostree-origin.c

```diff
@@ -468,9 +495,77 @@ rpmostree_origin_remove_packages (RpmOstreeOrigin  *origin,
     }
 
   if (changed)
-    update_keyfile_pkgs_from_cache (origin, FALSE);
+    update_keyfile_pkgs_from_cache (origin, "packages", "requested",
+                                    origin->cached_packages, FALSE);
   if (local_changed)
-    update_keyfile_pkgs_from_cache (origin, TRUE);
+    update_keyfile_pkgs_from_cache (origin, "packages", "requested-local",
+                                    origin->cached_local_packages, TRUE);
 
   return TRUE;
 }
+
+gboolean
+rpmostree_origin_add_overrides (RpmOstreeOrigin  *origin,
+                                char            **packages,
+                                gboolean          is_remove_override,
+                                GError          **error)
+{
+  /* that's all we support for now */
+  g_assert (is_remove_override);
+
+  gboolean changed = FALSE;
+  for (char **it = packages; it && *it; it++)
+    {
+      const char *pkg = *it;
+      if (g_hash_table_contains (origin->cached_overrides_remove, pkg))
+        return glnx_throw (error, "Package '%s' is already removed", pkg);
+
+      g_hash_table_add (origin->cached_overrides_remove, g_strdup (pkg));
+      changed = TRUE;
+    }
+
+  if (changed)
+    update_keyfile_pkgs_from_cache (origin, "overrides", "remove",
+                                    origin->cached_overrides_remove, FALSE);
+
+  return TRUE;
+}
+
+gboolean
+rpmostree_origin_remove_overrides (RpmOstreeOrigin  *origin,
+                                   char            **packages,
+                                   GError          **error)
+{
+  gboolean changed = FALSE;
+  for (char **it = packages; it && *it; it++)
+    {
+      const char *pkg = *it;
+      if (!g_hash_table_contains (origin->cached_overrides_remove, pkg))
+        return glnx_throw (error, "No overrides for package '%s'", pkg);
+
+      g_hash_table_remove (origin->cached_overrides_remove, pkg);
```

Comment:
That makes way more sense. I made a separate commit for it to make use of that in the whole file.